### DISCUSSION TITLE
Dice value changes depending on Ignore_index pixel values

### DISCUSSION THREAD

**Discussion by CuriousMonkey7** [Lightning-AI/torchmetrics#1524] [Classification]
I have been using torchmetric to calculate Dice, JaccardIndex and Accuracy with `ignore_index` for a segmentation problem. Ideally pixel value of ignore_index pixels should not affect the metrics. But when I change the pixel value of ignore_index pixel in the predicted mask, Dice changes. 
Also, in all three metrics, when `average="macro` ignore_index class is considered while taking the average, so even if the pixel is exact the same(other than ignore_index pixels), the metric value is less than 1.
Code:
```python
from torchmetrics import Dice, Accuracy, JaccardIndex

metrics = {
    "macro_dice": Dice(average="macro", num_classes=3, ignore_index =0),
    "acc": Accuracy(average="macro",task='multiclass',num_classes=3, ignore_index =0),
    "iou": JaccardIndex(average="macro",task="multiclass", num_classes=3, ignore_index =0),
    
}
def get_metrics(y_pred, y_true):
    macro_dice = metrics["macro_dice"](y_pred, y_true)
    acc = metrics["acc"](y_pred, y_true)
    iou = metrics["iou"](y_pred, y_true)
    return macro_dice, iou, acc

```
Example 1
```python
y_pred = torch.tensor([1,1,1,2])
y_true = torch.tensor([1,1,0,2])
print(get_metrics(y_pred, y_true))
> (tensor(0.9000), tensor(0.6667), tensor(0.6667))
```
Example 2
```python
y_pred = torch.tensor([1,1,2,2])
y_true = torch.tensor([1,1,0,2])
print(get_metrics(y_pred, y_true))
> (tensor(0.8333), tensor(0.6667), tensor(0.6667))
```

Is this a bug, or I'm doing something wrong?
